File: src/LegacyVault.Android/app/src/main/kotlin/com/legacyvault/app/data/remote/dto/CategoryDto.kt
package com.legacyvault.app.data.remote.dto

import kotlinx.serialization.Serializable

@Serializable
data class CategoryDto(
    val id: String,
    val type: String,           // "General" | "Vault" — mapped to CategoryType in domain
    val name: String,
    val icon: String,
    val isEncrypted: Boolean,
    val encryptionSalt: String? = null,
    val passwordHint: String? = null,
    val isFavorite: Boolean,
    val pageCount: Int,
    val createdAt: String,
    val updatedAt: String
)

@Serializable
data class CreateCategoryRequest(
    val name: String,
    val icon: String = "📁",
    val type: String = "General",       // "General" | "Vault"
    val isEncrypted: Boolean = false,
    val encryptionSalt: String? = null,
    val passwordHint: String? = null
)

@Serializable
data class UpdateCategoryRequest(
    val name: String,
    val icon: String,
    val passwordHint: String? = null
)

File: src/LegacyVault.API/DTOs/Categories/CategoryDto.cs
using LegacyVault.API.Models.Enums;

namespace LegacyVault.API.DTOs.Categories;

public class CategoryDto
{
    public Guid Id { get; set; }
    public CategoryType Type { get; set; }
    public string Name { get; set; } = string.Empty;
    public string Icon { get; set; } = string.Empty;
    public bool IsEncrypted { get; set; }
    public string? EncryptionSalt { get; set; }
    public string? PasswordHint { get; set; }
    public bool IsFavorite { get; set; }
    public int PageCount { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }
}
